projects/arkis/index.js

```diff
@@ -0,0 +1,100 @@
+const { getLogs } = require("../helper/cache/getLogs");
+const { sumTokens2 } = require("../helper/unwrapLPs")
+
+const infoAbi = require("./agreementAbi.json");
+const config = require("./config.json");
+
+
+/**
+ * Fetches logs from a factory contract on a specified blockchain.
+ *
+ * @param {Object} api - The API instance to interact with the blockchain.
+ * @param {string} chain - The name of the blockchain to fetch logs from.
+ * @param {string} factoryType - The type of factory to fetch logs for. Can be "agreement" or "account".
+ * @returns {Promise<Array<string>>} - A promise that resolves to an array of log addresses.
+ */
+async function fetchFactoryLogs(api, chain, factoryType) {
+  const topic = factoryType === "agreement" ? "AgreementCreated(address)" : "AccountDeployed(address)";
+  const eventAbi = factoryType === "agreement" ? "event AgreementCreated(address agreement)" : "event AccountDeployed(address indexed account)";
+
+  try {
+    const logs = await getLogs({
```

Comment:
add extra key here, there is a bug in our cache function, it creates one cache per address if `extraKey` field is missing

Reply:
ah, scratch that, in your case, topic field is optional/not needed, we derive it from eventAbi